domain/src/test/java/com/github/dodobest/domain/CalculatorTest.kt

```diff
@@ -8,48 +8,178 @@ import org.junit.runner.RunWith
-                IllegalArgumentException::class.java
-            ) { calculator.evaluate(inputString) }
-            assertThat(thrown).isInstanceOf(IllegalArgumentException::class.java)
-        }
-        else -> {
-            throw IllegalArgumentException("지원하지 않는 기댓값을 입력했습니다")
-        }
+    @Test
+    fun test_plus_calculation() {
+        // when : 사용자가 덧셈 식을 입력하면
+        val actual: Double = calculator.evaluate("1+2+3")
+
+        // then : 올바른 덧셈 결과를 계산한다
+        assertThat(actual).isEqualTo(6)
+    }
+
+    @Test
+    fun test_minus_calculation() {
+        // when : 사용자가 뺄셈 식을 입력하면
+        val actual: Double = calculator.evaluate("10-2-3")
+
+        // then : 올바른 뺄셈 결과를 계산한다
+        assertThat(actual).isEqualTo(5)
+    }
+
+    @Test
+    fun test_multiple_calculation() {
+        // when : 사용자가 곱셈 식을 입력하면
+        val actual: Double = calculator.evaluate("10*-5*9")
+
+        // then : 올바른 곱셈 결과를 계산한다
+        assertThat(actual).isEqualTo(-450)
+    }
+
+    @Test
+    fun test_divide_calculation() {
+        // when : 사용자가 나누어 떨어지는 나눗셈 식을 입력하면
+        val actual: Double = calculator.evaluate("120/2/3")
+
+        // then : 올바른 나눗셈 결과를 계산한다
+        assertThat(actual).isEqualTo(20)
+    }
+
+    @Test
+    fun test_not_divided_calculation() {
+        // when : 사용자가 나누어 떨어지지 않는 나눗셈 식을 입력하면
+        val actual = calculator.evaluate("10/3")
+
+        // then : 올바른 나눗셈 결과를 계산한다
+        assertThat(actual).isWithin(1.0e-5).of(3.3333333)
+    }
+
+    @Test
+    fun test_all_arithmetic_input_calculation() {
+        // when : 사용자가 사칙연산을 모두 포함하는 사칙연산 식을 입력하면
+        val actual: Double = calculator.evaluate("2 + 3 * 4 / 2")
+
+        // then : 올바른 계산 결과를 계산한다
+        assertThat(actual).isEqualTo(10)
+    }
+
+    @Test
+    fun test_positive_num_sign_calculation() {
+        // when : +가 붙은 양수를 입력하면
+        val actual: Double = calculator.evaluate("+10+10/+20")
+
+        // then : 올바른 계산 결과를 계산한다
+        assertThat(actual).isEqualTo(1)
+    }
+
+    @Test
+    fun test_null_input_calculation() {
+        // when : 사용자가 사칙연산 식에 null을 입력하면
+        val thrown: IllegalArgumentException = assertThrows(
+            IllegalArgumentException::class.java
+        ) { calculator.evaluate("120+null/3") }
+
+        // then : IllegalArgumentException 을 발생시킨다
+        assertThat(thrown).isInstanceOf(IllegalArgumentException::class.java)
+        assertThat(thrown).hasMessageThat().contains("의도한 Exception")
+
     }
 
+    @Test
+    fun test_blank_input_calculation() {
+        // when : 사용자가 사칙연산 식에 빈 문자열을 입력하면
+        val thrown: IllegalArgumentException = assertThrows(
+            IllegalArgumentException::class.java
+        ) { calculator.evaluate("120/10- *3+ /1") }
+
+        // then : IllegalArgumentException 을 발생시킨다
+        assertThat(thrown).isInstanceOf(IllegalArgumentException::class.java)
+        assertThat(thrown).hasMessageThat().contains("의도한 Exception")
```

Comment:
테스트 코드만 보고도 어떤 일이 발생하는지 설명서 처럼 알아야하는데, Exception 메시지가 모두 "의도한 Exception" 이 포함된다고만 하는 것은 의미가 없습니다.

특정 시점에 특정 Exception이 발생하는 것이므로, 해당 Exception에 대한 메시지를 하드코딩 해주시면 됩니다.

```suggestion
        assertThat(thrown).hasMessageThat().isEqualTo("[의도한 Exception]빈 공백문자를 입력했습니다.")
        // isEqualTo 나 contains나 무방합니다만, 완전히 똑같은 Exception 메시지를 넣어주세요.
```